### PR TITLE
Add example of tune calculation (postprocessing)

### PR DESCRIPTION
**Pull request by cemitch99**
An example of the calculation of tunes using postprocessing of tracking data.

The lattice is a periodic FODO channel for which the tune = phase advance per period / (2pi) is exactly known.

The horizontal tune of a single particle is computed and compared against the predicted value.  
Calculation is performed using both 4-turn formulae and using PyNAFF, https://pypi.org/project/PyNAFF/.

First attempt to address issue #391

### CODE REVIEW

File: examples/fodo_tune/README.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
+.. _examples-fodo-tune:
+
+Tune Calculation in a Periodic FODO Channel
+============================================
```
Comment (ax3l) [inline examples/fodo_tune/README.rst]:
```suggestion
Tune Calculation in a Periodic FODO Channel
===========================================
```

File: examples/fodo_tune/README.rst [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+* :math:`\beta_\mathrm{y} = 2.82161941` m
+* :math:`\alpha_\mathrm{y} = 1.59050035`
+
+We use a 2 GeV electron beam with initial unnormalized rms emittance of 2 nm.
+
+The horizontal tune of a single particle is obtained from period-by-period tracking data using several algorithms.
+
+In this test, the computed horizontal tune must agree with the nominal value to within acceptable tolerance.
+
+This example requires installation of PyNAFF:  https://pypi.org/project/PyNAFF/
```
Comment (ax3l) [inline examples/fodo_tune/README.rst]: I checked the requirements on PyNAFF: https://github.com/nkarast/PyNAFF/blob/master/requirements.txt

They are a subset of ours, so we could easily make it a mandatory dependency for Python builds in our `requirements.txt` :)

Comment (cemitch99) [reply examples/fodo_tune/README.rst]: That's fantastic.  I have some ideas about a few additional testing/details to add to the script, so it isn't finalized (but it works for me with PyNAFF installed).

Also, it's still worth exploring independently some methods to calculate the tunes in-situ.

**Comment by ax3l** [comment]
Tests: there seems to be one more value in `vars['TWIN']` than in `vars['ZTABS']`

File: examples/fodo_tune/README.rst [hunk 3]
```diff
@@ -0,0 +1,59 @@
+.. _examples-fodo-tune:
+
+Tune Calculation in a Periodic FODO Channel
+===========================================
+
+This is identical to the FODO example, except that tracking for 100 periods is used to extract the horizontal tune.
```
Comment (ax3l) [inline examples/fodo_tune/README.rst]:
```suggestion
This is identical to the :ref:`FODO example <examples-fodo>`, except that tracking for 100 periods is used to extract the horizontal tune.
```

File: examples/fodo_tune/README.rst [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+* :math:`\beta_\mathrm{y} = 2.82161941` m
+* :math:`\alpha_\mathrm{y} = 1.59050035`
+
+We use a 2 GeV electron beam with initial unnormalized rms emittance of 2 nm.
+
+The horizontal tune of a single particle is obtained from period-by-period tracking data using several algorithms.
+
+In this test, the computed horizontal tune must agree with the nominal value to within acceptable tolerance.
+
+This example requires installation of PyNAFF:  https://pypi.org/project/PyNAFF/
```
Comment (ax3l) [inline examples/fodo_tune/README.rst]:
```suggestion
This example requires installation of `PyNAFF <https://github.com/nkarast/PyNAFF>`__.
```

**Review by ax3l — APPROVED** [review]
This is awesome, thank you!

**Pull request merged**